implementation/src/main/java/io/smallrye/health/checks/UrlHealthCheck.java

```diff
@@ -0,0 +1,123 @@
+package io.smallrye.health.checks;
+
+import java.net.HttpURLConnection;
+import java.net.Socket;
+import java.net.URL;
+
+import org.eclipse.microprofile.health.HealthCheck;
+import org.eclipse.microprofile.health.HealthCheckResponse;
+import org.eclipse.microprofile.health.HealthCheckResponseBuilder;
+
+/**
+ * Health check implementation to check if host is reachable using a Http URL connection.
+ *
+ * <pre>
+ * {@code
+ * &#64;Produces
+ * &#64;ApplicationScoped
+ * &#64;Liveness
+ * HealthCheck check1() {
+ *   return new UrlHealthCheck("www.google.com");
+ * }
+ * }
+ * </pre>
+ *
+ * @see Socket
+ */
+public class UrlHealthCheck implements HealthCheck {
+
+    static final String DEFAULT_NAME = "Url Check";
+    static final int DEFAULT_TIMEOUT = 2000;
+    static final String DEFAULT_REQUEST_METHOD = "GET";
+    static final int DEFAULT_EXPECTED_STATUS_CODE = HttpURLConnection.HTTP_OK;
+
+    private String url;
+    private String name;
+    private int timeout;
+    private String requestMethod;
+    private int statusCode;
+
+    public UrlHealthCheck(String url) {
+        this.url = url;
+        this.requestMethod = DEFAULT_REQUEST_METHOD;
+        this.statusCode = DEFAULT_EXPECTED_STATUS_CODE;
+        this.name = DEFAULT_NAME;
+        this.timeout = DEFAULT_TIMEOUT;
+    }
+
+    @Override
+    public HealthCheckResponse call() {
+        final HealthCheckResponseBuilder healthCheckResponseBuilder = HealthCheckResponse
+                .named(name);
+        healthCheckResponseBuilder.withData("host", String.format("%s %s", this.requestMethod, this.url));
+        try {
+            final HttpURLConnection httpUrlConn = (HttpURLConnection) new URL(this.url)
+                    .openConnection();
+
+            httpUrlConn.setRequestMethod(requestMethod);
+
+            httpUrlConn.setConnectTimeout(timeout);
+            httpUrlConn.setReadTimeout(timeout);
+
+            final boolean isUp = httpUrlConn.getResponseCode() == statusCode;
+
+            if (!isUp) {
+                healthCheckResponseBuilder.withData("error", String.format("Expected response code %d but actual is %d",
+                        statusCode, httpUrlConn.getResponseCode()));
+            }
+
+            healthCheckResponseBuilder.state(isUp);
+
+        } catch (Exception e) {
+            healthCheckResponseBuilder.withData("error", String.format("%s not found", e.getMessage()));
```

Comment:
I guess this should be just the error message or this.url + "not found". You are catching here also MalformedURLException for instance.

Reply:
Well, notice that there is a field called `host` where we are putting al this information. So the error is just the error and the host is placed in another field for this reason I decided to just put `e.getMessage`.

Reply:
Ok, I have added the exception type and the message. The reason is that the message when the exception is an `UnkownHostException` is not clear of what is happening.